src/main/java/com/github/manolo8/darkbot/gui/MapDrawer.java

```diff
@@ -122,613 +50,172 @@ public MapDrawer(Main main) {
-        Graphics2D g2 = setupDraw(g);
 
+        mapGraphics.setup(g, getWidth(), getHeight());
 
         synchronized (Main.UPDATE_LOCKER) {
-            drawZones(g2);
-            if (hasFlag(DisplayFlag.ZONES)) drawCustomZones(g2);
-            drawInfos(g2);
-            drawHealth(g2);
-            drawTrail(g2);
-            drawStaticEntities(g2);
-            drawDynamicEntities(g2);
-            drawHero(g2);
+            onPaint();
         }
 
-        if (config.BOT_SETTINGS.OTHER.DEV_STUFF) {
-            g2.setFont(cs.FONTS.TINY);
-            g2.setColor(cs.TEXT_DARK);
-            synchronized (Main.UPDATE_LOCKER) {
-                List<Entity> entities = mapManager.entities.allEntities.stream().flatMap(Collection::stream)
-                        .filter(e -> e.id > 150_000_000 && e.id < 160_000_000 || e instanceof Mine)
-                        .filter(e -> e.locationInfo.isLoaded())
-                        .collect(Collectors.toList());
-
-                g2.setColor(cs.TEXTS_BACKGROUND);
-                for (Entity e : entities) {
-                    Location loc = e.locationInfo.now;
-                    int strWidth = g2.getFontMetrics().stringWidth(e.toString());
-                    g2.fillRect(translateX(loc.getX()) - (strWidth >> 1), translateY(loc.getY()) - 7, strWidth, 8);
-                }
-                g2.setColor(cs.TEXT);
-                g2.setFont(cs.FONTS.TINY);
-                for (Entity e : entities) {
-                    Location loc = e.locationInfo.now;
-                    drawString(g2, e.toString(), translateX(loc.getX()), translateY(loc.getY()), Align.MID);
-                }
-            }
-        }
-
-        synchronized (Main.UPDATE_LOCKER) {
-            if (!drawGroup(g2)) drawBoosters(g2);
-        }
-
-        if (hasFlag(DisplayFlag.STATS_AREA))
-            drawBackgroundedText(g2, Align.LEFT,
-                    "cre/h " + STAT_FORMAT.format(statsManager.earnedCredits()),
-                    "uri/h " + STAT_FORMAT.format(statsManager.earnedUridium()),
-                    "exp/h " + STAT_FORMAT.format(statsManager.earnedExperience()),
-                    "hon/h " + STAT_FORMAT.format(statsManager.earnedHonor()),
-                    "cargo " + statsManager.deposit + "/" + statsManager.depositTotal,
-                    "death " + guiManager.deaths + '/' + (config.GENERAL.SAFETY.MAX_DEATHS > -1 ? config.GENERAL.SAFETY.MAX_DEATHS : "∞"));
-
-        if (hovering && main.config.BOT_SETTINGS.MAP_DISPLAY.MAP_START_STOP) drawActionButton(g2);
-    }
-
-    protected Graphics2D setupDraw(Graphics g) {
-        cs = main.config.BOT_SETTINGS.MAP_DISPLAY.cs;
-
-        height = getHeight();
-        width = getWidth();
-        mid = width / 2;
-
-        g.setColor(cs.BACKGROUND);
-        g.fillRect(0, 0, width, height);
-
-        Graphics2D g2 = (Graphics2D) g.create();
-
-        g2.setRenderingHints(hints);
-        return g2;
+        //mapGraphics.dispose();
```

Comment:
```suggestion
        mapGraphics.dispose();
```

Keep the call to mapGraphics.dispose(), even if it ends up being an empty method that does nothing
i think it's reasonable to assume we may need to free some resources after drawing, so we can leave the call there